app/lib/llm_clients/openai/client.rb

```diff
@@ -1,77 +1,74 @@
-# rubocop:disable all
+require "openai"
+
 module LlmClients
   module Openai
-    class Client < Client
-      def call(prompt)
-        @stats = new_stats
-        request = request(prompt)
+    class Client < LlmClients::Client
+      include ChatRequestHelper
+      include EmbeddingRequestHelper
+      include CompletionRequestHelper
 
-        Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == "https") do |http|
-          http.request(request) do |response|
-            raise response_error_for(response) unless response.code.to_i >= 200 && response.code.to_i < 300
+      NETWORK_TIMEOUT = 8
 
-            stats[:start_time] = current_time
-            stats[:first_token_time] = current_time
-            body = response.read_body
-            content = ""
-            each_message(body) do |message|
-              @stats[:tokens] += 1
-              content += message if message
-            end
+      def complete(prompt, &)
+        complete_request(prompt, &)
+      end
 
-            yield content
+      def chat(message, &)
+        chat_request(ChatMessageHelper.new(message).chat_history, &)
```

Comment:
Oh, you're calling `new` anyway; do you need the `include`s?

Reply:
See above re helpers. I was using two patterns but calling them both helpers. In this case this is a helper proper, and I think this usage is correct _and_ unrelated to the various modules I was including.